SAT/Z3Interfacing.cpp

```diff
@@ -436,8 +436,10 @@ struct EvaluateInModel
     } else if (expr.is_app()) {
       auto f = expr.decl();
       auto vfunc = self._fromZ3.get(f);
```

Comment:
Again, `evaluatedArgs` might be `nullptr` in some cases. Assert that it isn't when we need them.